# Loading the pretrianed RF Model

In [8]:
import joblib

optimized_GB_model = joblib.load('optimized_GB_model.pkl')

In [9]:
optimized_GB_model

GradientBoostingClassifier(learning_rate=0.5, n_estimators=50, random_state=42)

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [11]:
from sklearn.datasets import make_moons
from sklearn.datasets import make_circles

def generate_dataset(sample=1000, random_state=42):   
    X, y = make_circles(n_samples=sample,noise=0.1, random_state=random_state)
    df1 = pd.DataFrame(X)
    df1['y'] = y
    
    X1, y1 = make_moons(n_samples=sample, noise=0.1, random_state=random_state)
    df2 = pd.DataFrame(X1)
    df2['y'] = y
    
    df = pd.concat([df1, df2])
    df = df.sample(frac=1).reset_index(drop=True)
    return df[[0,1]].to_numpy(), df.y

# production data - use this in the new notebook with the saved model
def generate_dataset_prod(sample=300, random_state=60):   
    X, y = make_circles(n_samples=sample,noise=0.1, random_state=random_state)
    df1 = pd.DataFrame(X)
    df1['y'] = y
    
    X1, y1 = make_moons(n_samples=sample, noise=0.1, random_state=random_state)
    df2 = pd.DataFrame(X1)
    df2['y'] = y
    
    df = pd.concat([df1, df2])
    df = df.sample(frac=1).reset_index(drop=True)
    return df[[0,1]].to_numpy(), df.y

In [12]:
X, y = generate_dataset_prod()

In [13]:
from sklearn.metrics import classification_report
print(classification_report(y, optimized_GB_model.predict(X)))

              precision    recall  f1-score   support

           0       0.85      0.86      0.85       300
           1       0.86      0.85      0.85       300

    accuracy                           0.85       600
   macro avg       0.85      0.85      0.85       600
weighted avg       0.85      0.85      0.85       600



The optimized gradient boosting model that was exported in the previous notebook and loaded into this notebook was able to achieve a test accuracy of 85%, which is similar to the one in the previous notebook. This shows the boosting model's ability to significantly outperform other models in terms of classification accuracy.